service-map/src/main/java/fi/nls/oskari/map/geometry/WKTHelper.java

```diff
@@ -78,16 +85,29 @@ public static String transform(final String wkt, final String sourceSRS, final S
     /**
      * @param wkt       original geometry in EPSG:4326
      * @param targetSRS "EPSG:3067"
-     * @return projected geometry as wkt
+     * @return null if:
+     *  - geometry is null
+     *  - geometry is not a Polygon
+     *  - any of the coordinates in the exterior ring is not within [-180,-90,180,90]
+     *  otherwise return the exterior ring of the polygon projected to the targetSRS and
+     *  possibly with extra interpolated points in-between of the original segments
      */
     public static String transformLayerCoverage(final String wkt, final String targetSRS) {
         GeometryFactory gf = new GeometryFactory();
         Geometry geom = parseWKT(wkt, gf);
-        if (geom == null) {
+        if (geom == null || !(geom instanceof Polygon)) {
             return null;
         }
         Polygon polygon = (Polygon) geom;
-        CoordinateSequence cs = GeometryHelper.interpolateLinear(polygon.getExteriorRing(), 1.0, gf);
+        LineString exterior = polygon.getExteriorRing();
+        boolean withinWGS84Bounds = GeometryHelper.isWithin(exterior.getCoordinateSequence(),
+                WGS84_LON_MIN, WGS84_LAT_MIN,
+                WGS84_LON_MAX, WGS84_LAT_MAX);
+        if (!withinWGS84Bounds) {
+            log.info("Layer coverage not within WGS84 bounds, not interpolating or transforming extent");
+            return null;
```

Comment:
Returning empty instead of unmodified geometry is better in this case since having the wrong information for the frontend will most likely make the layer be invisible to the user (not shown on the map properly) while having no information will just prevent the user from zooming to the extent (since the extent is not known).